client/src/pages/Home/index.js

```diff
@@ -1,24 +1,65 @@
-import { useEffect, useState } from "react";
+import React, { useEffect, useState } from "react";
 
 import { Pagination, ResourceList } from "../../components";
 import { useSearchParams } from "../../hooks";
 import { ResourceService, useService } from "../../services";
 
-export function Home() {
+const Home = () => {
 	const resourceService = useService(ResourceService);
 	const searchParams = useSearchParams();
-	const [{ lastPage, resources } = {}, setEnvelope] = useState();
+	const [{ lastPage, resources } = {}, setEnvelope] = useState({});
+
+	const storedPageSize = localStorage.getItem("pageSizePreference");
+	let initialPageSize;
+
+	try {
+		initialPageSize = storedPageSize ? JSON.parse(storedPageSize) : "default";
+	} catch (error) {
+		// Handle the error as needed
+		initialPageSize = "default";
+		throw new Error("Error parsing page size preference:", error);
+	}
+
+	const [pageSize, setPageSize] = useState(initialPageSize);
+
+	// This function is to save the selected page-size into local storage
+	const savePageSizePreference = (pageSize) => {
+		localStorage.setItem("pageSizePreference", JSON.stringify(pageSize));
+	};
```

Comment:
💡 This is a lot of code in the middle of the home page to handle something relatively minor (and that might also be needed _elsewhere_). Don't just jam everything into one file, think about how you can separate out unrelated concerns to make your code easier to read (and test). For example, what if you extracted a _service_ to handle local storage, or maybe a _hook_:
```suggestion
const [pageSize, setPageSize] = useLocalStorage("pageSizePreference", "default");
```
Also note some of my other suggestions (error handling, comments\*) apply here too.

\* _if you're going to put comments on functions, use [JSDoc](https://jsdoc.app/) so they're structured and useful_